sixtyfps_compiler/generator/cpp.rs

```diff
@@ -1768,7 +1768,7 @@ fn compile_expression(expr: &llr::Expression, ctx: &EvaluationContext) -> String
             _ => panic!("Expression::ObjectAccess's base expression is not an Object type"),
         },
         Expression::ArrayIndex { array, index } => {
-            format!("[&](const auto &model, const auto &index){{ model->track_row_data_changes(index); return model->row_data(index); }}({}, {})", compile_expression(array, ctx), compile_expression(index, ctx))
+            format!("[&](const auto &model, const auto &index){{ model->track_row_data_changes(index); const auto v = model->row_data(index); if (v.has_value()) return *v; else return {{}}; }}({}, {})", compile_expression(array, ctx), compile_expression(index, ctx))
```

Comment:
It's fascinating that this works nowadays in C++, without the need to specify the return type - especially the `{}` part. Very cool :)

Reply:
Yes, I was also rather surprised by that.

But let's see: Maybe the CI will uncover that I did something wrong there:-)

Reply:
See? Ci says it does not work:-/ I wonder why it did for me... maybe I used a different compiler?

Reply:
Maybe this could be done by "re-exporting" the type parameter for `Model` as `template <typename ModelData> class Model { public: typedef ModelData ValueType; ... }` and then use `decltype(model)::Valuetype` ?

Or perhaps this could be a lambda with explicit template parameters? `[&]<ModelData>(const sixtyfps::Model<ModelData> &model, const auto &index) -> ModelData { ... }` ?